dashboard/modules/job/job_head.py

```diff
@@ -35,9 +37,21 @@
 def _init_ray_and_catch_exceptions(f: Callable) -> Callable:
     @wraps(f)
     async def check(self, *args, **kwargs):
-        if not ray.is_initialized():
-            ray.init(address="auto", namespace=RAY_INTERNAL_JOBS_NAMESPACE)
         try:
+            if not ray.is_initialized():
```

Comment:
Why the dashboard needs to init ray? It makes the dashboard unstable with C++ core worker initialized. I think we should stick some rules to make dashboard and dashboard agent clean and maintainable,
- Use single thread with asyncio. (not mixing multi-threading with asyncio)
- A pure Python process without ray C++ core worker.
- Modules not coupling each other.

Reply:
The dashboard initializes Ray so it can use public APIs/functionality for runtime_env setup, cleanup, etc. & process management

Reply:
> 

I think it's better to implement untime_env setup, cleanup, etc without Ray initialized. Clean process state makes it stable and maintainable.

Reply:
I think we're far from "clean process state" in the dashboard right now. If anything, I think we should be treating the dashboard _more_ like a regular Ray driver instead of relying on a bunch of private APIs and direct connections/queries to the GCS & redis